.ci/generate_bear_requirements.py

```diff
@@ -144,6 +154,47 @@ def get_pip_requirements(requirements):
     return _get_requirements(requirements, '~=', inherited_requirements)
 
 
+def deep_update(target, src):
+    for key, value in src.items():
+        if not key in target:
+            target[key] = copy.deepcopy(value)
+        else:
+            if isinstance(value, list):
+                target[key].extend(value)
+            elif isinstance(value, dict):
+                deep_update(target[key], value)
+            elif isinstance(value, set):
+                target[key].update(value.copy())
+            else:
+                target[key] = copy.copy(value)
+
+
+def deep_diff(target, src):
+    errors = []
+    for key, value in src.items():
+        if not key in target:
+                errors.append((key, 'Missing'))
+        elif target[key] != value:
+            if isinstance(value, list):
+                if [x for x in value if x not in target[key]]:
+                    errors.append(key)
+            elif isinstance(value, dict):
+                if target[key] != value:
+                    errors.append((key, deep_diff(target[key], value)))
+            elif isinstance(value, set):
+                if set(target[key]).symmetric_difference(value):
+                    errors.append(key)
+            else:
+                errors.append((key, target[key]))
+    return errors
+
+
+def sort_requirements(req_dict):
+    for reqs in ['pip_requirements', 'npm_requirements', 'gem_requirements']:
```

Comment:
`... in req_dict.keys():`

and `reqs` is plural, which is inappropriate for a string.
plural should be used for variables which hold containers of multiple items.

Reply:
No for `... in req_dict.keys()` as there may be other keys like `overrides` that we dont want to sort.

Will change the variable name to `key`, sounds fine?